mslice/models/cut/cut_algorithm.py

```diff
@@ -90,10 +90,15 @@ def _compute_cut_nonPSD(selected_workspace, cut_axis, integration_axis, emode):
     extents = " ,".join(map(str, (xdim.getMinimum(), xdim.getMaximum())))
 
     # Hack to (deep) copy log data (ExperimentInfo)
-    _tmpws = ConvertToMD(ws_out, EnableLogging=False, StoreInADS=False, PreprocDetectorsWS='-')
```

Comment:
Could you help me understand what is happening here? I'm struggling a little..

Reply:
Yeah... it's a bit over complicated. The main issue is to be able to copy the sample logs from `ws_out` which is a `Workspace2D` to the final output which is an `MDHistoWorkspace`. The cleanest solution would be to use the `CopyLogs` algorithm and use that but at the moment `CopyLogs` does not work on MD workspaces. So, instead I'm using the MDWorkspace's `copyExperimentInfos` method but this requires another MDWorkspace. This is created using `ConvertToMD` but for `dEAnalysisMode='Indirect'`, `ConvertToMD` requires each detector to have the fixed `E_f` defined. This is the case for `nxs` files but not for `nxspe` files. (`SofQW` allows you to specify a value for `E_f` to override that in the detectors / or in case the values in the detectors don't exist, but `ConvertToMD` doesn't).

So the main solutions are:

1. Use this complicated work-around / hack which creates a blank `MatrixWorkspace`, copies the logs, converts it to MD and then uses `copyExperimentInfos`. The only advantage of this is that it is pure python and doesn't need changes to the C++ code (and would allow MSlice master to be backwards compatible with older Mantid versions).
2. Change `CopyLogs` to allow it to copy logs from `MatrixWorkspace` to MD workspaces.
3. Change `ConvertToMD` to allow specifying a value for `E_f`
4. Change the `copyExperimentInfos` method to allow it to accept a `MatrixWorkspace` too.

If we don't like the current implementation, I think changing `CopyLogs` is the best option. It would also allow some other python code in Mantid (mostly in various workflow algorithms) to be refactored.

Reply:
I see. Thanks for the explanation. I'm happy keeping this as a solution for the release but I agree I think the better long-term solution is to update `mantid`  probabaly with all 3 of your final points above.
Could you add this comment to that code so we don't forget what is a happening?